src/main/java/sg/dex/starfish/impl/memory/AMemoryOperation.java

```diff
@@ -19,7 +23,33 @@
 public abstract class AMemoryOperation extends AMemoryAsset implements Operation {
 
     protected AMemoryOperation(String metaString, MemoryAgent memoryAgent) {
-        super(metaString, memoryAgent);
+        super(getMetaWithProv(metaString,memoryAgent), memoryAgent);
+    }
+
+    /**
+     * Method to add provenace in the metadata if not present for all Memory Operation
+     * @param metaString Metadata as string
+     * @param memoryAgent agent
+     * @return metadata with PROVENANCE
+     */
+    private static String getMetaWithProv(String metaString,MemoryAgent memoryAgent) {
+
+        Map<String,Object> meta = JSON.toMap(metaString);
+        //check if already provenance is present
+
+        if(meta.get(Constant.PROVENANCE)== null) {
```

Comment:
What is the logic here? Please add comments describing why something is being done. Ideally reference DEP12